src/engine/execute.js

```diff
@@ -148,19 +140,19 @@ var execute = function (sequencer, thread) {
             execute(sequencer, thread);
             if (thread.status === Thread.STATUS_PROMISE_WAIT) {
                 return;
-            } else {
-                // Execution returned immediately,
-                // and presumably a value was reported, so pop the stack.
-                currentStackFrame.waitingReporter = null;
-                thread.popStack();
             }
+
+            // Execution returned immediately,
+            // and presumably a value was reported, so pop the stack.
+            currentStackFrame.waitingReporter = null;
+            thread.popStack();
         }
         argValues[inputName] = currentStackFrame.reported[inputName];
     }
 
     // Add any mutation to args (e.g., for procedures).
-    var mutation = blockContainer.getMutation(currentBlockId);
-    if (mutation) {
+    var mutation = blockContainer.getMutation(block);
+    if (mutation !== null) {
```

Comment:
I think this one's OK: I think it's more likely that `getMutation` will disappear as a function than change to something which can return `undefined`. In general, though, I'd be more comfortable avoiding micro-optimizations that trade safety for speed unless we can show (say, through profiling) that the impact is non-trivial.

That said...
> What we really need to a good refactor of the runtime engine to reduce the number of function calls as they are proving very expensive.

👍